syntax = "proto3";

package silenteight.datasource.api.date;

import "google/api/annotations.proto";
import "silenteight/datasource/api/common/common.proto";

option java_package = "com.silenteight.datasource.api.date";
option java_outer_classname = "DateInputProto";
option java_multiple_files = true;

service DateInputService {
  rpc BatchGetMatchDateInputsV1 (silenteight.datasource.api.common.BatchGetMatchAgentInputsRequest) returns (stream BatchGetMatchDateInputsV1Response) {
    option (google.api.http) = {
      post: "/agent-inputs/date/versions/1/matches:batchGet",
      body: "*"
    };
  }
}

message BatchGetMatchDateInputsV1Response {
  repeated DateInput date_inputs = 1;
}

message DateInput {
  string match = 1;
  repeated DateFeatureInput date_feature_inputs = 2;
}

message DateFeatureInput {
  string feature = 1;
  repeated string alerted_party_dates = 2;
  repeated string watchlist_dates = 3;
}
